linker/shared/src/main/scala/org/scalajs/linker/StandardLinker.scala

```diff
@@ -46,6 +46,9 @@ object StandardLinker {
         LinkerBackendImpl(backendConfig))
   }
 
+  def clearable(config: Config): ClearableLinker =
+    new ClearableLinker(() => apply(config), config.batchMode)
+
```

Comment:
In that case, shouldn't we make the constructor of `ClearableLinker` private[linker]? IIUC the idea here was that they can share a config.

Reply:
If we do that, there's no way to create a `ClearableLinker` wrapping a non-standard linker, is there?

Reply:
Ah, indeed, we should allow this. Although wondering if it should be part of the super stable API :S

Maybe we should move the implementation of `ClearableLinker` to `standard._` as well and just have an abstract class in `linker`?

Reply:
`ClearableLinker` only relies on a `() => Linker` function to create its linker, so only super-stable stuff. If we move it to `standard._`, and some API receives a `() => Linker`, it could not make a `ClearableLinker` out of it without relying on non-super-stable API.

For example, an alternative build plugin could be configured with a `() => Linker` rather than directly with a `ClearableLinker`. That build plugin might want to wrap that `() => Linker` inside a `ClearableLinker`, but to do that it would have to rely on `standard._` (currently sbt-scalajs only uses `linker._`).

Maybe we could make the constructor `private` and add an `apply` method in `ClearableLinker`'s companion object instead of exposing the constructor. And even make `ClearableLinker` a `trait`, that simply inherits from `Linker` and adds an abstract `def clear(): Unit`. The companion's `apply` would return an instance of a private class implementing that trait.